src/mpyl/utilities/parallel/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""Utility for running commands in parallel"""
+from concurrent.futures import Future, ThreadPoolExecutor
+from dataclasses import dataclass
+from typing import Callable, Any, TypeVar, Iterable
+
+T = TypeVar("T")
+
+
+@dataclass(frozen=True)
+class ParallelCommand:
+    function: Callable
+    parameters: dict[str, Any]
+
+
+def run_in_parallel(
+    commands: list[ParallelCommand],
+    number_of_threads: int,
+) -> list[T]:
+    threads: list[Future] = []
+    executor = ThreadPoolExecutor(max_workers=number_of_threads)
+    for command in commands:
+        threads.append(executor.submit(command.function, **command.parameters))
+
+    results: list[T] = list(__flatten([thread.result() for thread in threads]))
+
+    return results
+
+
+def __flatten(object_to_flatten: Any):
```

Comment:
Is this more readable than the itertools.chain?

Reply:
I don't remember the details, but there was a reason I used this. Not sure if `itertools.chain` didn't do precisely what we need or Sam commented something

Reply:
could be that the object is not always an iterable? I would then move the if to the outside and call itertools.chain